src/store/reducers/tasks.js

```diff
@@ -0,0 +1,20 @@
+const tasksReducer = (state = [], action) => {
+  switch (action.type) {
+    case 'ADD_TASK':
+      return [...state, action.payload];
+    case 'REMOVE_TASK':
+      return state.filter(({ id }) => id !== action.payload);
+    case 'TOGGLE_DONE': {
+      const index = state.findIndex(({ id }) => id === action.payload);
+      return [
+        ...state.slice(0, index),
```

Comment:
This is fine, but I would use `map` to do it, for me it's simpler to read

`state.map((item, index) => index === indexToUpdate ? {
    ...item,
    done: !item.done
} : item)`